app/lib/features/pins/pages/create_pin_page.dart

```diff
@@ -0,0 +1,345 @@
+import 'dart:io';
+
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/input_text_field.dart';
+import 'package:acter/common/widgets/md_editor_with_preview.dart';
+import 'package:acter/common/widgets/side_sheet.dart';
+import 'package:acter/common/widgets/spaces/select_space_form_field.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter/features/pins/pin_utils/pin_utils.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:mime/mime.dart';
+
+class CreatePinPage extends ConsumerStatefulWidget {
+  final String? initialSelectedSpace;
+  static const titleFieldKey = Key('create-pin-title-field');
+  static const contentFieldKey = Key('create-pin-content-field');
+  static const urlFieldKey = Key('create-pin-url-field');
+  static const submitBtn = Key('create-pin-submit');
+
+  const CreatePinPage({super.key, this.initialSelectedSpace});
+
+  @override
+  ConsumerState<CreatePinPage> createState() => _CreatePinSheetConsumerState();
+}
+
+class _CreatePinSheetConsumerState extends ConsumerState<CreatePinPage> {
+  final TextEditingController _titleController = TextEditingController();
+  final TextEditingController _linkController = TextEditingController();
+  final TextEditingController _textController = TextEditingController();
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      final spaceNotifier = ref.read(selectedSpaceIdProvider.notifier);
+      spaceNotifier.state = widget.initialSelectedSpace;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final attachments = ref.watch(selectedPinAttachmentsProvider);
+    return SideSheet(
```

Comment:
Ahh, the wording goes of this widget like this but this doesn't include internal mechanism to open as sideheet. This is fine here, the actual changes regarding that are in the routes part where we use custom transition to determine how it should open. I should change this widget name.

Reply:
Yeah but it seems like this has no actual affect on how it will present itself if router custom transition is used. Regardless this'd need to be addressed.

Reply:
Can't we change routing for create pin flow in order to get rid of side sheet flow?

Reply:
Changed the ```SideSheet``` widget to ```SliverScaffold``` and removed its side-sheet properties. That doesn't affect other sidesheet pages as they use ```SideSheetPage``` as route transition.